kod/object/active/holder/room/monsroom/survivalroom.kod

```diff
@@ -1295,6 +1464,43 @@ messages:
    {
       return piLevel;
    }
+   
+   GetHasBegun()
+   {
+      return piHasBegun;
+   }
+
+   SomeoneSaid(what = $,type = $,string = $,parm1 = $,parm2 = $,parm3 = $,
+               parm4 = $, parm5 = $,parm6 = $,parm7 = $,parm8 = $)
+   {
+      if NOT isClass(what,&player) OR type = SAY_YELL
+      {
+         propagate;
+      }
+
+      if StringEqual(string,"leave survival")
+      {
+         Send(self,@BootUser,#who=what);
```

Comment:
Need to post this otherwise there are issues when it propagates and the user is no longer on the same screen. Alternatively adding return; after booting the user may work also.